lib/pecorino/adapters/base_adapter.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+# An adapter allows Pecorino throttles, leaky buckets and other
+# resources to interfact to a data storage backend - a database, usually.
+class Pecorino::Adapters::BaseAdapter
+  # Returns the state of a leaky bucket. The state should be a tuple of two
+  # values: the current level (Float) and whether the bucket is now at capacity (Boolean)
+  def state(key:, capacity:, leak_rate:)
+    [0, false]
+  end
+
+  # Adds tokens to the leaky bucket. The return value is a tuple of two
+  # values: the current level (Float) and whether the bucket is now at capacity (Boolean)
+  def add_tokens(key:, capacity:, leak_rate:, n_tokens:)
+    [0, false]
+  end
+
+  # Adds tokens to the leaky bucket conditionally. If there is capacity, the tokens will
+  # be added. If there isn't - the fillup will be rejected. The return value is a triplet of
+  # the current level (Float), whether the bucket is now at capacity (Boolean)
+  # and whether the fillup was accepted (Boolean)
+  def add_tokens_conditionally(key:, capacity:, leak_rate:, n_tokens:)
+    [0, false, false]
+  end
+
+  # Sets a timed block for the given key - this is used when a throttle fires. The return value
+  # is not defined - the call should always succeed.
+  def set_block(key:, block_for:)
+  end
+
+  # Returns the time until which a block for a given key is in effect. If there is no block in
```

Comment:
**nitpick:** it's cool that you're writting these comments, but would it be better to have yard definitions here instead?

Reply:
Why not both? 😆 good one